### DISCUSSION TITLE
Accessing quadrature points of a feature to modify material properties

### DISCUSSION THREAD

**Discussion by rayaprolu143** [idaholab/moose#21066] [Q&A Modules: Phase field] [answered]
Hello,

To modify the material properties within a feature, I want to access the quadrature points of elements that are part of a feature (e.g., precipitate, bubble). I did try the method `getVarToFeatureVector(_current_elem->id())` of the `FeatureFloodCount` to know if an element has any features, but I only ended up getting `invalid id` for every element. Could anyone provide me guidance to achieve the latter?

Thanks,
Sreekar

**Comment by GiudGiud** [comment, marked as answer]
@laagesen @SudiptaBiswas

**Reply by SudiptaBiswas** [reply]
`getVarToFeatureVector` gives you the `grain_id` to order parameter map for the element. If the element is within a bubble, you would get `invalid_id`.  Can you not identify the features by the active order parameter (or variable) values? We would need more information about what you are trying to implement to guide you toward the right approach.
